generators/php/sdk/src/endpoint/request/WrappedEndpointRequest.ts

```diff
@@ -176,9 +233,136 @@ export class WrappedEndpointRequest extends EndpointRequest {
+                                    this.writeMultipartPart(
+                                        writer,
+                                        ref,
+                                        `${this.context.getPropertyName(file.key.name)}`
+                                    );
+                                    writer.endControlFlow();
+                                } else {
+                                    this.writeMultipartPart(
+                                        writer,
+                                        ref,
+                                        `${this.context.getPropertyName(file.key.name)}`
+                                    );
+                                }
+                            },
+                            fileArray: (fileArray) => {
+                                if (fileArray.isOptional) {
+                                    const ref = `${this.getRequestParameterName()}->${this.context.getPropertyName(
+                                        file.key.name
+                                    )}`;
+                                    writer.controlFlow("if", php.codeblock(`${ref} != null`));
+                                    this.writeMultipartPartArray(
+                                        writer,
+                                        `${this.context.getPropertyName(file.key.name)}`
+                                    );
+                                    writer.endControlFlow();
+                                } else {
+                                    this.writeMultipartPartArray(
+                                        writer,
+                                        `${this.context.getPropertyName(file.key.name)}`
+                                    );
+                                }
+                            },
+                            _other: () => undefined
+                        }),
+                    bodyProperty: (bodyProperty) => {
+                        const propertyName = this.context.getPropertyName(bodyProperty.name.name);
+                        let ref = `${this.getRequestParameterName()}->${propertyName}`;
+                        let propType = bodyProperty.valueType;
+                        const isOptional = this.context.isOptional(propType);
+
+                        if (isOptional) {
+                            writer.controlFlow("if", php.codeblock(`${ref} != null`));
+                            propType = this.context.dereferenceOptional(propType);
+                        }
+
+                        const isCollection = this.context.isCollection(propType);
+                        if (isCollection) {
+                            writer.controlFlow("foreach", php.codeblock(`${ref} as $element`));
+                            ref = "$element";
+                            propType = this.context.dereferenceCollection(propType);
+                        }
```

Comment:
Discussed offline - we might be able to clean this up with a separate interface that captures each variable in a convenient structure. Not required at all, but something to explore a bit.

Reply:
Pulled out the value assignment generation but I think I still want to keep the block writing explicit so that it's obvious how it works for now and when we actually want to support something more sophisticated we can.